packages/utils/shared/deep-map.ts

```diff
@@ -1,25 +1,25 @@
 import { isObjectLike, isPlainObject } from 'lodash-es';
 
-export function deepMap(object: any, iterator: (value: any, key: string | number) => any, context?: any): any {
-	if (Array.isArray(object)) {
-		return object.map(function (val, key) {
-			return isObjectLike(val) ? deepMap(val, iterator, context) : iterator.call(context, val, key);
-		});
-	} else if (isPlainObject(object)) {
-		const res: Record<string, any> = {};
+/**
+ * Maps over all values in an array or object recursively and calls the callback on primitive values.
+ * The called object will not be mutated. The callback will not be called for primitive values.
+ *
+ * @example
+ * deepMap({a: 1, b: [2]}, (value) => String(value)) // {a: "1", b: ["2"]}
+ */
+export function deepMap(object: any, callback: (value: any, key: string | number) => any, context?: any): any {
+	return recurse(object);
 
-		for (const key in object) {
-			const val = object[key];
-
-			if (isObjectLike(val)) {
-				res[key] = deepMap(val, iterator, context);
-			} else {
-				res[key] = iterator.call(context, val, key);
-			}
+	function recurse(value: any, key?: string | number): any {
+		if (Array.isArray(value)) {
+			return value.map(recurse);
+		} else if (isObjectLike(value) && isPlainObject(value)) {
```

Comment:
Are we able to use `isObject` here instead?

Reply:
Because `isObject(new Date()) === true` which we don't want here.